deployments/k8s/asyncapi-event-gateway/templates/deployment.yaml

```diff
@@ -0,0 +1,98 @@
+{{ $fileIsURL := (not (empty (.Values.env.EVENTGATEWAY_ASYNC_API_DOC | urlParse).host )) }}
+{{- if empty .Values.asyncapiFileContent }}
+  {{- if not $fileIsURL }}
+    {{ fail "Please either set the AsyncAPI file content via --set-file asyncapi-event-gateway.asyncapiFileContent=<filename> or a valid URL via --set asyncapi-event-gateway.env.EVENTGATEWAY_ASYNC_API_DOC=<url>" }}
+  {{- end }}
+{{- end }}
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "asyncapi-event-gateway.fullname" . }}
+  labels:
+    {{- include "asyncapi-event-gateway.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "asyncapi-event-gateway.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "asyncapi-event-gateway.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      volumes:
+      {{- if not $fileIsURL }}
+        - name: asyncapi-file-volume
+          configMap:
+            name: {{ include "asyncapi-event-gateway.fullname" . }}
+      {{- end }}
+      {{- with .Values.volumes }}
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          volumeMounts:
+          {{- if not $fileIsURL }}
+            - name: asyncapi-file-volume
+              mountPath: {{ .Values.asyncapiFileMountPath }}
+              subPath: asyncapi.yaml
+          {{- end }}
+          {{- with .Values.volumeMounts }}
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          env:
+          {{- range $name, $value := .Values.env }}
+            - name: {{ $name | quote }}
+              value: {{ $value | quote }}
+          {{- end }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.ports.healthcheck }}
+              protocol: TCP
+            - name: websocket
+              containerPort: {{ .Values.ports.websocket }}
+              protocol: TCP
+            {{- range .Values.ports.brokers }}
+            - name: "broker-{{ . }}"
+              containerPort: {{ . }}
+              protocol: TCP
+            {{ end }}
+          livenessProbe:
+            httpGet:
+              path: /
+              port: http
```

Comment:
Would there be benefit in having a liveness script that hits `/ws` to see if it's returning a 101 status code?  That might be a better indication.  I think wget might be in the Docker image already, and I'm sure that can be leveraged.

Reply:
Actually, I think `curl` is in the image, as well.  I just put together a basic example of how this might work.

The config would have to change to something like:

```yaml
    livenessProbe:
      exec:
        command:
          - /go/websocket_handshake_probe.sh http://localhost:3000
```

See: https://gist.github.com/kevinswiber/a0fd2d9e8b327d6b8f9a5a7bfa097085

Reply:
I created such issue so we can work on it later https://github.com/asyncapi/event-gateway/issues/77